### PR TITLE
Test networking configuration

### PR DESCRIPTION
**Pull request by ermanno**
### Description of changes
Test networking configuration.

### Tests
Run tests on EC2. Recipe/test is skipped on Docker because `sysctl` doesn't work there.

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [x] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/recipes/controls/aws_parallelcluster_config/networking_spec.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+
+control 'networking_configured' do
+  title 'Check that networking has been configured'
+
+  only_if { !os_properties.virtualized? }
```
Comment (enrico-usai) [inline test/recipes/controls/aws_parallelcluster_config/networking_spec.rb:15]: You can remove this check, we already have this kind of tests for example for sticky bits, let's see if it works for all OSes.

File: cookbooks/aws-parallelcluster-config/recipes/networking.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+#
+# Cookbook:: aws-parallelcluster
+# Recipe:: networking
+#
+# Copyright:: 2013-2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (enrico-usai) [inline cookbooks/aws-parallelcluster-config/recipes/networking.rb]: nitpick: 2023

File: cookbooks/aws-parallelcluster-config/recipes/networking.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the
+# License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+return if virtualized?
```
Comment (enrico-usai) [inline cookbooks/aws-parallelcluster-config/recipes/networking.rb:18]: You can remove this check, this work for sure (see sticky_bits resource/recipe).

File: test/recipes/controls/aws_parallelcluster_config/networking_spec.rb (hunk 1 above)
Comment (ermanno) [reply test/recipes/controls/aws_parallelcluster_config/networking_spec.rb:15]: Are you sure? From what I can see, in the [sticky_bits_spec](https://github.com/aws/aws-parallelcluster-cookbook/blob/redhat8/test/resources/controls/aws_parallelcluster_config/sticky_bits_spec.rb#L4) we use `!os_properties.virtualized?`

File: cookbooks/aws-parallelcluster-config/recipes/networking.rb (hunk 3 above)
Comment (ermanno) [reply cookbooks/aws-parallelcluster-config/recipes/networking.rb:18]: From what I can see, though, in the sticky bits case we skip if we are on Docker (see [comment above](https://github.com/aws/aws-parallelcluster-cookbook/pull/1734#discussion_r1100076651)).

**Review by enrico-usai — DISMISSED** [review]

**Review by enrico-usai — APPROVED** [review]

**Pull request merged**